src/test/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisServiceFactorsCountTest.java

```diff
@@ -0,0 +1,473 @@
+        );
+
+        assertEquals(0L, result.variableCount());
+        assertEquals(0L, result.resultCount());
+        verifyNoInteractions(filterService, actionsService);
+    }
+
+    @Test
+    void testEmptyContingenciesListDoesNotCallActionsService() {
+        SensitivityAnalysisParametersInfos parameters = createParameters(
+                List.of(createInjectionsSet(true, List.of(BRANCH1_UUID), List.of())),
+                null, null, null, null
+        );
+
+        mockFilterServiceResponse(Map.of("monitored-0", 1L));
+
+        factorCountService.getFactorCount(
+                NETWORK_UUID, VARIANT_ID,
+                parameters.getSensitivityInjectionsSet(),
+                parameters.getSensitivityInjection(),
+                parameters.getSensitivityHVDC(),
+                parameters.getSensitivityPST(),
+                parameters.getSensitivityNodes()
+        );
+
+        verifyNoInteractions(actionsService);
+    }
+
+    @Test
+    void testAllTypes() {
+        SensitivityAnalysisParametersInfos parameters = createParameters(
+                List.of(createInjectionsSet(true, List.of(BRANCH1_UUID, BRANCH2_UUID),
+                        List.of(CONTINGENCY1_UUID))),
+                List.of(createInjection(true, List.of(BRANCH1_UUID), List.of(GEN1_UUID, GEN2_UUID),
+                        List.of(CONTINGENCY1_UUID, CONTINGENCY2_UUID))),
+                List.of(createHVDC(true, List.of(BRANCH2_UUID), List.of(HVDC1_UUID), List.of())),
+                List.of(createPST(true, List.of(BRANCH1_UUID), List.of(PST1_UUID), List.of())),
+                List.of(createNodes(true, List.of(VOLTAGE_LEVEL1_UUID),
+                        List.of(EQUIPMENT_REGULATION1_UUID), List.of(CONTINGENCY1_UUID)))
+        );
+
+        mockFilterServiceResponse(Map.of(
+                "monitored-0", 2L,  // injections set
+                "monitored-1", 1L,  // injections
+                "variables-1", 2L,
+                "monitored-2", 1L,  // hvdc
+                "variables-2", 1L,
+                "monitored-3", 1L,  // pst
+                "variables-3", 1L,
+                "monitored-4", 2L,  // nodes
+                "variables-4", 3L
+        ));
+
+        mockContingencyServiceResponse(Map.of(
+                "contingencies-0", 1L,  // injections set: 1 contingency
+                "contingencies-1", 2L,  // injections: 2 contingencies
+                "contingencies-4", 1L   // nodes: 1 contingency
+        ));
+
+        FactorCount result = factorCountService.getFactorCount(
+                NETWORK_UUID, VARIANT_ID,
+                parameters.getSensitivityInjectionsSet(),
+                parameters.getSensitivityInjection(),
+                parameters.getSensitivityHVDC(),
+                parameters.getSensitivityPST(),
+                parameters.getSensitivityNodes()
+        );
+
+        assertEquals(8L, result.variableCount());
+        assertEquals(36L, result.resultCount());
```

Comment:
I would appreciate to see the way 36 is calculated here with a message like "3 monitored × 2 variables × 1 (base) × 2 (empirical factor)"
it would make debugging easier if this test happened to fail